malloc/malloc.c

```diff
@@ -13,7 +14,9 @@
 */
 char *string_dup(char *src)
 {
-
+    char *dup = (char *) malloc(strlen(src));
```

Comment:
This is an interesting technique, is it necessary to cast the malloc'd memory on the right side of the assignment?